src/ripple/protocol/SField.h

```diff
@@ -526,13 +542,18 @@ extern SF_ACCOUNT const sfUnauthorize;
 extern SF_ACCOUNT const sfRegularKey;
 extern SF_ACCOUNT const sfNFTokenMinter;
 extern SF_ACCOUNT const sfEmitCallback;
+extern SF_ACCOUNT const sfAMMAccount;
```

Comment:
Is the `sfAMMAccount` field really necessary? Is there a reason that the `sfAccount` field couldn't be reused?

Reply:
Ok, yeah, that makes sense.

Referring back to my comment about #4361 (https://github.com/XRPLF/rippled/pull/4294#discussion_r1139323892), check out the function `isOwnedByAccount` (the current name) in `RPCHelpers.h`. The second check is to see if there's an `sfAccount` field, and if so, check it against the `accountID` parameter. That is one reason why most objects associated with an account have that `sfAccount` field set to the object owner. Is it fair to call that a standard?

It's probably a pain in the neck, to keep the useful goal of having different field names for `ltAMM` and the nested `sfAuctionSlot` field, I think it would make more sense to
1. Get rid of `sfAMMAccount`, and use `sfAccount` in the `ltAMM` object.
2. Create a new `SF_ACCOUNT const sfDiscountAccount` (or a better name, or _maybe_ reuse `sfAuthorize`) and use that in the nested `sfAuctionSlot` and `sfAuthAccounts` fields.

This would completely sidestep the problem of needing to update one of this PR or #4361 after the other is merged.

Reply:
I'm going to rename `sfAMMAccount` and keep other fields as `sfAccount`. Will also rename `amm_account` to `account` in `amm_info`.